dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/impl/ConfigAccessorBuilderImpl.java

```diff
@@ -0,0 +1,197 @@
+            this.converter = converter;
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessorBuilder<T> withDefault(T value) {
+        synchronized (this) {
+            this.defaultValue = value;
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessorBuilder<T> withStringDefault(String value) {
+        synchronized (this) {
+            this.defaultString = value;
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessorBuilder<T> cacheFor(long value, ChronoUnit timeUnit) {
+        if (timeUnit == null) {
+            //TODO NLS
+            throw new IllegalArgumentException("timeUnit may not be null");
+        }
+        if (value < 0) {
+            //TODO NLS
+            throw new IllegalArgumentException("cacheFor value must be non-negative");
+        }
+        synchronized (this) {
+            this.cacheFor = value;
+            this.cacheForUnit = timeUnit;
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessorBuilder<T> evaluateVariables(boolean evaluateVariables) {
+        synchronized (this) {
+            this.evaluateVariables = evaluateVariables;
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessorBuilder<T> addLookupSuffix(String suffixValue) {
+        if (suffixValue == null) {
+            //TODO NLS
+            throw new IllegalArgumentException("suffixValue may not be null");
+        }
+        synchronized (this) {
+            this.suffixValues.add(suffixValue);
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ConfigAccessor<T> build() {
+        ConfigAccessor<T> accessor;
+        synchronized (this) {
+            Duration cacheForDuration = null;
+            if (cacheFor > 0) {
+                cacheForDuration = Duration.of(cacheFor, cacheForUnit);
+            }
+
+            List<String> propertyNames = generatePropertyNameList(this.propertyName, this.suffixValues);
+
+            accessor = new ConfigAccessorImpl<T>(this.config, propertyNames, this.conversionType, cacheForDuration, this.evaluateVariables, this.defaultValue, this.defaultString, this.converter);
+        }
+        return accessor;
+    }
+
+    /**
+     * This method uses a binary count down to append suffixes to a base propertyName.
+     *
+     * e.g.
+     * if there are 4 suffixes then the counter starts at 15 (1111) which would mean that all of the suffixes are applied in order.
+     * The counter then drops to 14 (1110) which would mean all but the last suffix are applied.... and so on until zero where no suffixes are applied
+     *
+     * if propertyName is "BASE" and the suffixes are ONE, TWO, THREE and FOUR then the output is (with counter for reference)
+     *
+     * (1111) BASE.ONE.TWO.THREE.FOUR
```

Comment:
@Emily-Jiang FYI, this is my current impl that works out the list of property names with suffixes